packages-exp/app-check-compat/src/service.ts

```diff
@@ -47,24 +49,42 @@ export class AppCheckService implements FirebaseAppCheck, _FirebaseService {
     } else {
       provider = new CustomProvider({ getToken: siteKeyOrProvider.getToken });
     }
-    initializeAppCheck(this.app, {
+    this._delegate = initializeAppCheck(this.app, {
```

Comment:
We should check if AppCheck is already initialized first. If it is, we will just get it through `provider.getImmediate()`; This can happen if both app-check and app-check-compat are included in an app and app-check is initialized first.

Reply:
hm, that doesn't seem to be a valid use case because people need to provide a provider to `activate`, so if they are using app-check-compat, they should use `activate` and never call `initializeAppCheck()` from the modular package.